sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -62,9 +64,19 @@ abstract class RDG extends UnaryExpression with ExpectsInputTypes with Stateful
  * Usually the random seed needs to be renewed at each execution under streaming queries.
  */
 trait ExpressionWithRandomSeed {
+  def seedExpression: Expression
   def withNewSeed(seed: Long): Expression
 }
 
+/**
+ * A place holder expression used in random functions, will be replaced after analyze.
+ */
+case class DefaultSeed() extends Expression with Unevaluable {
```

Comment:
`case object UnresolvedSeed extends Expression with Unevaluable`?

Reply:
yes, it's better.